docs/bgp-policy.md

```diff
@@ -0,0 +1,190 @@
+# BGPPolicy
+
+## Table of Contents
+
+<!-- toc -->
+- [What is BGPPolicy?](#what-is-bgppolicy)
+- [Prerequisites](#prerequisites)
+- [The BGPPolicy resource](#the-bgppolicy-resource)
+  - [NodeSelector](#nodeselector)
+  - [LocalASN](#localasn)
+  - [ListenPort](#listenport)
+  - [Advertisements](#advertisements)
+  - [BGPPeers](#bgppeers)
+- [BGP router ID](#bgp-router-id)
+- [Usage examples](#usage-examples)
+  - [Combined Advertisements of Service, Pod, and Egress IPs](#combined-advertisements-of-service-pod-and-egress-ips)
+  - [Advertise Egress IPs to external BGP peers with more than one hops](#advertise-egress-ips-to-external-bgp-peers-with-more-than-one-hops)
+- [Limitations](#limitations)
+<!-- /toc -->
+
+## What is BGPPolicy?
+
+`BGPPolicy` is a custom resource that allows users to initialize BGP process on selected Kubernetes Nodes and advertise
+Service IPs, Pod IPs, and Egress IPs to remote BGP peers, facilitating the integration of Kubernetes cluster with external
+BGP-enabled network.
+
+## Prerequisites
+
+BGPPolicy was introduced in Antrea v2.1 as an alpha feature. A feature gate, `BGPPolicy` must be enabled on antrea-agent
+in the `antrea-config` ConfigMap for the feature to work, like the following:
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: antrea-config
+  namespace: kube-system
+data:
+  antrea-agent.conf: |
+    featureGates:
+      BGPPolicy: true
+```
+
+## The BGPPolicy resource
+
+A BGPPolicy in Kubernetes is a Custom Resource Definition (CRD) object. Like all CRD objects, you can POST a BGPPolicy
+definition to the API server to create a new instance.
+
+The following manifest creates a BGPPolicy object. It can start BGP process with ASN `64512`, listening on port `179`,
+on Nodes labeled with `bgp=enabled`. The process will advertise ClusterIPs, LoadBalancerIPs, and ExternalIPs to a BGP
+peer at IP address `192.168.77.200`, which has ASN `65001` and listens on port `179`:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: BGPPolicy
+metadata:
+  name: example-bgp-policy
+spec:
+  nodeSelector:
+    matchLabels:
+      bgp: enabled
+  localASN: 64512
+  listenPort: 179
+  advertisements:
+    service:
+      ipTypes:
+        - ClusterIP
+        - ExternalIP
+        - LoadBalancerIP
```

Comment:
nit: I think it would be more concise / readable to use `ipTypes: [ClusterIP, ExternalIP, LoadBalancerIP]`